routing/routes.go

```diff
@@ -27,6 +27,24 @@ func NewRoutes() (r *Routes) {
 			"/api",
 			handlers.Root,
 		},
+		Route{
+			"FailExample",
+			"GET",
+			"/api/500",
+			handlers.FailExample,
+		},
+		Route{
+			"ServiceList",
+			"GET",
+			"/api/namespaces/{id:[a-z]+}/services",
```

Comment:
I have 2 thoughts here:
1. I know I am repeating myself in different PRs on this topic, but what about openshift? openshift uses projects, not namespaces, at least towards the users, and I think that our app is considered to be a user.. unless we use this only internally on both openshift and k8s, so it might be ok (not sure if namespaces are "hidden" in openshift api or not, it used to be accessible a while ago)
2. having an api named "namespaces" as a main resource name implies (at least to me) that we (sws) have an entity named "namespace". which is not exactly correct
3. where is the option of viewing all service accross all namespaces? is it possible via the current api?

Reply:
I would maintain namespace(s) at API level and change to "project" in the UI as we discussed.
Openshift maps in one to one project and namespace.
And, there is visibility in the operations, so you can't filter pods or services per project using a command, you will use a "namespace", i.e. --all-project doesn't exist, but you will use --all-namespaces or -n <namespace> using your oc get ... command.

So, as API is using the same terminology, my thoughts is we should follow that, changing here to project is confusing.

But I agree, let's use project at UI level as it's the same concept.

Reply:
@lucasponce isn't the API of oc/kubectl focusing on the entity (like service) and the namespace is passed as a parameter? (-n)

Reply:
> @lucasponce isn't the API of oc/kubectl focusing on the entity (like service) and the namespace is passed as a parameter? (-n)

It is also a first class entity, using the kubernetes go API:

```
in.k8s.CoreV1().Namespaces().List(emptyListOptions)
```

Then when you want to query a particular service:

```
service, err := in.k8s.CoreV1().Services(namespace).Get(serviceName, emptyGetOptions)
```

Reply:
@lucasponce yes namespace is also a first class entity. but as you also write above, when querying for *service* - service is the one that is first class entity with namespace parameter.

Reply:
1 - I would go for consistency on our application. If we show projects to users, our API should respond to projects. Then namespaces are something connected to the technical implementation of the solution, which IMO should be independent from what we show to the world.
2 - @abonas would it make sense if we use project instead? Would be correct to have a resource called project?
3- Good point. I wasn't thinking about this scenario actually. Is the UI going to need that? If so, I should add a `/api/services` endpoint. @mtho11 @aljesusg what are your thoughts? Actually, I don't know exactly how it is going to be the UI we want to achieve :sweat_smile:

Reply:
I still thinking at API level we should move in namespace for clarity, but in the UI we can use project, as a project is a superset of the namespace entity.
If you perform a "oc describe service <my-service>" you won't get a Project but a Namespace, and that's what I guess it makes sense to maintain in the REST API layer.
Also, to be consistent with the OpenShift API you can take as example here
https://docs.openshift.com/container-platform/3.7/rest_api/api/v1.Namespace.html
So a endpoint to get all projects sounds weird and we don't needed in the REST layer.
But as commented, in the UI, yes, it makes sense to use the same terminology as the OpenShift Console.

Reply:
- lets start with namespaces, and if needed, add project later.
even for the simple fact that querying for project on k8s env if our app is deployed there, will result with a not so nice error :)
- lets start with services endpoint

Reply:
We can change the name of the endpoints later, we talked about in https://issues.jboss.org/browse/SWS-69, for have sense have /namespace or /ns (short url) if the client is called Namespaces.